File: viajes/src/gestorAplicacion/adminVuelos/Tiquete.java
package gestorAplicacion.adminVuelos;
import java.io.Serializable;
import gestorAplicacion.hangar.*;


import gestorAplicacion.alojamiento.Alojamiento;

/*LA CLASE TIQUETE CONTIENE LA INFORMACION ASOCIADA AL VUELO, SILLA Y ALOJAMIENTOS SELECCIONADOS
TAMBIEN CONTIENE UN ID PARA TIQUETE Y EL VALOR DEL PRECIO.
SUS METODOS NOS PERMITEN CONSOLIDAR EL RESUMEN DEL TIQUETE QUE TIENE UN PASAJARO ASOCIADO A UN VUELO*/
public class Tiquete implements Serializable {
	//ATRIBUTOS
	private static final long serialVersionUID = 1L;
	private int id;
	private int precio;
	private Vuelo vuelo;
	private Silla silla;
	private Pasajero pasajero;
	private Alojamiento alojamiento;

	//CONSTRUCTORES

	public Tiquete(int id, int precio, Vuelo vuelo) {
		this.id = id;
		this.precio = precio;
		this.vuelo = vuelo;
		vuelo.getTiquetes().add(this);
	}

	public Tiquete(int id, int precio, Vuelo vuelo, Silla silla, Pasajero pasajero, Alojamiento alojamiento) {
		this.id = id;
		this.precio = precio;
		this.vuelo = vuelo;
		this.silla = silla;
		this.pasajero = pasajero;
		this.alojamiento = alojamiento;
		vuelo.getTiquetes().add(this);
	}

	// METODOS

	/*ESTE METODO NO TIENE PARAMETROS DE ENTRADA PERO RETORNA UN BOOLEANO, YA QUE 
	  SU OBJETIVO ES ASIGNARLE EL PRECIO A CADA INSTANCIA DE TIQUETE OBTENIENDO LOS PRECIOS
	  DEL VUELO Y SILLA SELECCIONADOS CON UN DESCUENTO POR CONFORME AL ATRIBUTO EDAD DEL PASAJERO*/
	public boolean asignarPrecio() {
		boolean hayDescuento = false;
		int precio_total=vuelo.getPrecio() + this.getSilla().getClase().getPrecio();
		if (pasajero.getEdad()<5) {
			hayDescuento = true;
			this.precio = (int) (precio_total - (precio_total*0.25));
		}else if (pasajero.getEdad()>5 && pasajero.getEdad()<=10){
			this.precio = (int) (precio_total - (precio_total*0.15));
			hayDescuento = true;
		}else {
			this.precio = precio_total;
		}
		return hayDescuento;
	}

	/*ESTE METODO SOBRECARGA EL METODO ASGINAR PRECIO RECIBIENDO COMO PARAMETRO DE ENTRADA
	 UN ENTERO Y SU RETORNO ES VACIO. SU FUNCION SUMARLE A CADA INSTANCIA DE TIQUETE EL PRECIO DEL ALOJAMIENTO
	 SELECCIONADO DE ACUERDO AL NUMERO DE DIAS INGRESADO JUNTO A LOS PRECIOS DEL VUELO Y SILLA*/
	public void asignarPrecio(int num_dias) {
		int precio_total=vuelo.getPrecio()+ alojamiento.calcularPrecio(num_dias) + this.getSilla().getClase().getPrecio();
		if (pasajero.getEdad()<5) {
			this.precio = (int) (precio_total - (precio_total*0.15));
		}else if (pasajero.getEdad()>5 && pasajero.getEdad()<=10){
			this.precio = (int) (precio_total - (precio_total*0.15));
		}else {
			this.precio = precio_total;
		}
	}
	
	/* EL METODO AGREGAR LA INSTANCIA DE TIQUETE CREADA AL ARRAY
	   DE TIQUETES QUE TIENE ASOCIADO AL VUELO SELCCIONADO, POR TAL RAZON NO RECIBE PARAMETRO Y SU
	   RETORNO ES VACIO */
	public void confimarCompra() {
		this.vuelo.getTiquetes().add(this);
	}

	/*VAMOS A UTILIZAR EL METEDO TOSTRING PARA IMPRIR EL RESUMEN DEL TIQUETE ADQUIRIDO
	  DEPENDIENDO SI EL TIQUETE TIENE O NO ASOCIADO UN ALOJOMIENTO */
	public String toString()
	{
		if (this.alojamiento==null) {
			return  "*************************************\n"+
					"      Su compra ha sido exitosa\n"+
					"    Gracias por confiar en nostros\n"+
					"*************************************\n"+

					"------------------------------------\n"+
					"      Tiquete No."+ this.id + "\n"+
					"------------------------------------\n"+
					"Nombre Pasajero: " + pasajero.nombre + "\n" +
					"Fecha: " + vuelo.getFecha_de_salida() + "\n" +
					"Vuelo: " + vuelo.getID() + "\n" +
					"Num Silla: " + silla.getNumero_de_silla() + " "  + silla.getUbicacion() + "\n" +
					"Origen: " + vuelo.getOrigen() + "\n" +
					"Destino: " + vuelo.getDestino() + "\n" +
					"Precio Total: " + this.getPrecio() + "\n" +
					"------------------------------------\n";


		}else {
			
			return  "*************************************\n"+
					"      Su compra ha sido exitosa\n"+
					"    Gracias por confiar en nostros\n"+
					"*************************************\n"+
					"------------------------------------\n"+
					"      Tiquete No."+ this.id + "\n"+
					"------------------------------------\n"+
					"Nombre Pasajero: " + pasajero.nombre + "\n" +
					"Fecha: " + vuelo.getFecha_de_salida() + "\n" +
					"Vuelo: " + vuelo.getID() + "\n" +
					"Silla: " + silla.getNumero_de_silla() + " - "  + silla.getUbicacion()  + "\n" +
					"Origen: " + vuelo.getOrigen() + "\n" +
					"Destino: " + vuelo.getDestino() + "\n" +
					"Alojamiento: " + alojamiento.getNombre() + "\n" +
					"Precio Total: " + this.getPrecio() + "\n" +
					"------------------------------------\n";

		}
	}

	//GETTERS Y SETTERS

	public int getId() {
		return id;
	}
	public void setId(int id) {
		this.id = id;
	}
	public int getPrecio() {
		return precio;
	}
	public void setPrecio(int precio) {
		this.precio = precio;
	}
	public Vuelo getVuelo() {
		return vuelo;
	}
	public void setVuelo(Vuelo vuelo) {
		this.vuelo = vuelo;
	}
	public Silla getSilla() {
		return silla;
	}
	public void setSilla(Silla silla) {
		this.silla = silla;
		silla.setEstado(false);
	}
	public Pasajero getPasajero() {
		return pasajero;
	}
	public void setPasajero(Pasajero pasajero) {
		this.pasajero = pasajero;
	}
	public Alojamiento getAlojamiento() {
		return alojamiento;
	}
	public void setAlojamiento(Alojamiento alojamiento) {
		this.alojamiento = alojamiento;
	}
}

File: carpeta/gestorAplicacion/adminVuelos/Tiquete.py
from tkinter.messagebox import NO
from gestorAplicacion.hangar.Aeronave import Aeronave
from gestorAplicacion.hangar.Avion import Avion
from gestorAplicacion.hangar.Avioneta import Avioneta
from gestorAplicacion.hangar.Clase import Clase
from gestorAplicacion.hangar.Silla import Silla
from gestorAplicacion.hangar.Ubicacion import Ubicacion
from gestorAplicacion.alojamiento.Alojamiento import Alojamiento

class Tiquete:


    #CONSTRUCTOR
    def __init__(self, id, precio, vuelo):
        self._id = id
        self._precio = precio
        self._vuelo = vuelo
        self._silla = None
        self._pasajero = None
        self._alojamiento = None
        vuelo.getTiquetes().append(self)

    # METODOS

    #	Este metodo no tiene parametros de entra o de salida porque el valor resultante
    #	  es guardado en el atributo precio de cada instancia
    def asignarPrecio(self,*args):
        hayDescuento = False
        precio_total =self._vuelo.getPrecio() + self.getSilla().getClase().value
        if self._pasajero.getEdad()<5:
            hayDescuento = True
            self._precio = int((precio_total - (precio_total*0.25)))
        elif self._pasajero.getEdad()>5 and self._pasajero.getEdad()<=10:
            self._precio = int((precio_total - (precio_total*0.15)))
            hayDescuento = True
        else:
            self._precio = precio_total

        if len(args) != 0 :

            self._precio += self._alojamiento.calcularPrecio(args[0]) 

        return hayDescuento



    def confimarCompra(self):
        self._vuelo.getTiquetes().append(self)

    def __str__(self):
        if self._alojamiento is None:
            return "*************************************\n"+ "      Su compra ha sido exitosa\n"+ "    Gracias por confiar en nostros\n"+ "*************************************\n"+ "------------------------------------\n"+ "      Tiquete No."+ str(self._id) + "\n"+ "------------------------------------\n"+ "Nombre Pasajero: " + self._pasajero.nombre + "\n" + "Fecha: " + self._vuelo.getFecha_de_salida() + "\n" + "Vuelo: " + str(self._vuelo.getID()) + "\n" + "Num Silla: " + str(self._silla.getNumero_de_silla()) + " " + self._silla.getUbicacion().value + "\n" + "Origen: " + self._vuelo.getOrigen() + "\n" + "Destino: " + self._vuelo.getDestino() + "\n" + "Precio Total: " + str(self.getPrecio()) + "\n" + "------------------------------------\n"


        else:

            return "*************************************\n"+ "      Su compra ha sido exitosa\n"+ "    Gracias por confiar en nostros\n"+ "*************************************\n"+ "------------------------------------\n"+ "      Tiquete No."+ str(self._id) + "\n"+ "------------------------------------\n"+ "Nombre Pasajero: " + self._pasajero.nombre + "\n" + "Fecha: " + self._vuelo.getFecha_de_salida() + "\n" + "Vuelo: " +str(self._vuelo.getID()) + "\n" + "Silla: " + str(self._silla.getNumero_de_silla()) + " - " + self._silla.getUbicacion().value + "\n" + "Origen: " + self._vuelo.getOrigen() + "\n" + "Destino: " + self._vuelo.getDestino() + "\n" + "Alojamiento: " + self._alojamiento.getNombre() + "\n" + "Precio Total: " + str(self.getPrecio()) + "\n" + "------------------------------------\n"


    #GETTERS Y SETTERS

    def getId(self):
        return self._id

    def setId(self, id):
        self._id = id

    def getPrecio(self):
        return self._precio

    def setPrecio(self, precio):
        self._precio = precio

    def getVuelo(self):
        return self._vuelo

    def setVuelo(self, vuelo):
        self._vuelo = vuelo

    def getSilla(self):
        return self._silla

    def setSilla(self, silla):
        self._silla = silla
        silla.setEstado(False)

    def getPasajero(self):
        return self._pasajero

    def setPasajero(self, pasajero):
        self._pasajero = pasajero

    def getAlojamiento(self):
        return self._alojamiento
        
    def setAlojamiento(self, alojamiento):
        self._alojamiento = alojamiento
